olive/cli/quantize.py

```diff
@@ -0,0 +1,133 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+# ruff: noqa: T201
+# ruff: noqa: RUF012
+
+import tempfile
+from argparse import ArgumentParser
+from copy import deepcopy
+from typing import Any, Dict
+
+from olive.cli.base import (
+    BaseOliveCLICommand,
+    add_accelerator_options,
+    add_dataset_options,
+    add_input_model_options,
+    add_logging_options,
+    add_remote_options,
+    is_remote_run,
+    save_output_model,
+    update_accelerator_options,
+    update_dataset_options,
+    update_input_model_options,
+)
+from olive.common.utils import set_nested_dict_value
+
+
+class QuantizeCommand(BaseOliveCLICommand):
+    _CONFIG_TEMPLATE = {
+        "input_model": {"type": "HfModel", "load_kwargs": {"attn_implementation": "eager"}},
+        "systems": {
+            "local_system": {
+                "type": "LocalSystem",
+                "accelerators": [{"device": "gpu", "execution_providers": ["CUDAExecutionProvider"]}],
+            }
+        },
+        "data_configs": [
+            {
+                "name": "default_data_config",
+                "type": "HuggingfaceContainer",
+                "load_dataset_config": {},
+                "pre_process_data_config": {},
+                "dataloader_config": {},
+                "post_process_data_config": {},
+            }
+        ],
+        "passes": {
+            "awq": {"type": "AutoAWQQuantizer"},
```

Comment:
I tried running with only awq which should not need a data config. but I get this error 
![image](https://github.com/user-attachments/assets/5666b724-2aac-4218-be2c-a7ff0a067bfd)

Reply:
So, one algorithm needs dataset and the other doesn't. I don't think there's an easy way to configure `ArgumentParser` which this kind of requirement. This might need some `ArgumentParser` trickery to make it work. Somehow, the `required` option of dataset will need to be toggled based on the user input.

Reply:
thats not possible since the arg parse is defined before the user input. you could just make it optional but print a failure while creating the run config. That's what i did for some options that were optional based on user input in the generate-adapter command https://github.com/microsoft/Olive/blob/d3070aa5da2bf77e30bee3df0ce6659f79c192e3/olive/cli/generate_adapter.py#L96

Reply:
Isn't that misleading. Documentation says one thing but when used as per the documentation, it reports a failure.

Reply:
you could provide your own help text if you want. I don't think there is an issue making some options conditionally required.